WooCommerce/Classes/System/WooTabNavigationController.swift

```diff
@@ -0,0 +1,43 @@
+import UIKit
+
+/// Navigation controller for a Woo tab, shown as the root view controller of one of the tab bar.
+/// The first view controller always has large title, while the following view controllers in the navigation stack do not have large title by default.
+/// The following view controllers can override `preferredLargeTitleDisplayMode` function to change the large title display mode.
+final class WooTabNavigationController: UINavigationController {
+    init() {
+        super.init(nibName: nil, bundle: nil)
+        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.largeTitles) {
+            navigationBar.prefersLargeTitles = true
+            delegate = self
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        ServiceLocator.featureFlagService.isFeatureFlagEnabled(.largeTitles) ? .default: StyleManager.statusBarLight
```

Comment:
nit:
```suggestion
        ServiceLocator.featureFlagService.isFeatureFlagEnabled(.largeTitles) ? .default : StyleManager.statusBarLight
```

Reply:
updated in https://github.com/woocommerce/woocommerce-ios/pull/3752/commits/e2ca1b1a61fe1b4b8fd47323ca64488ff547225c!